js/query.js

```diff
@@ -1,70 +1,452 @@
+                        'gen-pos': coords,
+                        '19gen-pos': hg19coords,
+                        'mm9gen-pos': mm9coords,
+                        'tax-id': data.taxid,
+                        'species': speciesObj[data.taxid],
+                        'other-names': names,
+                        'gene-type': data.type_of_gene,
+                        'wikipedia': wiki,
+                        'omim': omim,
+                        'ensembl': ensembl,
+                        'vega': vega,
+                        'uniprot': uniprot,
+                        'pfam': pfam,
+                        'pharmgkb': pharmgkb,
+                        'prosite': prosite,
+                        'uniprot-summary': null
+                    });
                 });
+            } else {
+                resolve({
+                    'entrez-summary': data.summary,
+                    'alias': aliases,
+                    'hgnc-id': hgnc,
+                    'location': data.map_location,
+                    'gen-pos': coords,
+                    '19gen-pos': hg19coords,
+                    'mm9gen-pos': mm9coords,
+                    'tax-id': data.taxid,
+                    'species': speciesObj[data.taxid],
+                    'other-names': names,
+                    'gene-type': data.type_of_gene,
+                    'wikipedia': wiki,
+                    'omim': omim,
+                    'ensembl': ensembl,
+                    'vega': vega,
+                    'uniprot': uniprot,
+                    'pfam': pfam,
+                    'pharmgkb': pharmgkb,
+                    'prosite': prosite,
+                    'uniprot-summary': null
+                });
+            }
+        });
+
+    }
+
+    function getUniprotSummary(id) {
+        return new Promise(function(resolve, reject) {
+            if (!id) {
+                reject();
+            }
+
+            var summary;
+
+            fetch('http://www.uniprot.org/uniprot/' + id + '.xml').then(function(response) {
+                if (response.status !== 200) {
+                    console.log('Looks like there was a problem. Status Code: ' + response.status);
+
+                    reject();
+                }
+
+                response.text().then(function(data) {
+                    var parsedXML = xmlParser.parseFromString(data, "text/xml");
+                    var comments = parsedXML.querySelectorAll('comment[type="function"]');
+
+                    for (var i = 0; i < comments.length; i++) {
+                        summary = comments[0].textContent;
+                    }
+                    resolve(summary);
+                })
+            }).catch(function(err) {
+                console.error('Fetch Uniprot Error', err);
+
+                reject();
+            });
+        });
+    }
+
+    // Display and hide section headings as appropriate.
+    function displayHeadings() {
```

Comment:
Replace these with generic `showAll()` and `hideAll()` functions. I mention this above.